src/AssetsPage/AssetsTable/index.jsx

```diff
@@ -2,18 +2,30 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import Table from '@edx/paragon/src/Table';
 import Button from '@edx/paragon/src/Button';
+import Modal from '@edx/paragon/src/Modal';
+import classNames from 'classnames';
+import { connect } from 'react-redux';
 
 import FontAwesomeStyles from 'font-awesome/css/font-awesome.min.css';
+import { deleteAsset } from '../../data/actions/assets';
+
 
 class AssetsTable extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
       displayAssetsList: [],
+      modalOpen: false,
+      assetToDelete: {},
+      elementToFocusOnModalClose: {},
     };
 
     this.onDeleteClick = this.onDeleteClick.bind(this);
+    this.deleteAsset = this.deleteAsset.bind(this);
     this.addDeleteButton = this.addDeleteButton.bind(this);
+    this.closeModal = this.closeModal.bind(this);
+
+    this.trashcanRefs = {};
```

Comment:
I need to store the references to the trashcan icons in order to handle focus. The object is keyed by asset.id, and the value is the reference to the icon.

Reply:
Since focus isn't being highlighted in the browser ideally, you can use this event listener to see which elements have focus after a keystroke.

document.addEventListener("keyup", function() {
    console.log(document.activeElement);
});

Just enter that in the developer console, and it should log which button has focus as you're tabbing through.

Reply:
:+1: good tip! I'll look into why the focus outline isn't showing - my suspicion is it has something to do with the way we're scoping Bootstrap styles.